src/reactjs/components/DeviceGrid/DeviceGrid.tsx

```diff
@@ -6,193 +6,169 @@
 import * as React from 'react'
 import { deviceColumnDefs, checkboxColumn, defaultDeviceGridProps } from './deviceGridConfig'
 
-import { isFunc } from '../shared/Utilities'
+import { isFalsy, isFunc } from '../shared/Utilities'
 
 // Below 3 imports to support Localization
-import { translateColumnDefs } from "../shared/Methods";
-import { PcsGrid } from '../shared/pcsGrid/PcsGrid';
-import { HttpClient } from '../services/HttpClient';
-import { DomainContext } from '../shared/context/BasicContextProvider';
+import { translateColumnDefs } from '../shared/Methods'
+import { PcsGrid } from '../shared/pcsGrid/PcsGrid'
+import { HttpClient } from '../services/HttpClient'
+import { DomainContext } from '../shared/context/BasicContextProvider'
 
-import { library } from "@fortawesome/fontawesome-svg-core";
-import * as Icons from "@fortawesome/free-solid-svg-icons";
+import { library } from '@fortawesome/fontawesome-svg-core'
+import * as Icons from '@fortawesome/free-solid-svg-icons'
 
 const iconList = Object.keys(Icons)
-  .filter((key) => key !== "fas" && key !== "prefix")
-  .map((icon) => Icons[icon]);
-library.add(...iconList);
-
+  .filter((key) => key !== 'fas' && key !== 'prefix')
+  .map((icon) => Icons[icon])
+library.add(...iconList)
 
 interface gridStates {
-    columnDefs: any,
-    rowData: any,
-    rowSelection: any,
-    autoGroupColumnDef: any,
-    softSelectedDevice: any
+  columnDefs: any
+  rowData: any
+  rowSelection: any
+  autoGroupColumnDef: any
+  softSelectedDevice: any
 }
 
 export interface gridProps {
-    deviceId: string;
-    mpsServer: string;
-    getSelectedDevices?: any;
-    selectedDevices?: any;
-    filter?: string;
+  deviceId: any
```

Comment:
why were these changed to any?

Reply:
Since deviceId and mpsServer values are taken from the URL while using webpack, the probable types are string and null. Updated now with proper null checks for all the files